### DISCUSSION TITLE
Ckan executable and config locations in Docker container

### DISCUSSION THREAD

**Discussion by frankyhollywood** [ckan/ckan#8534] [Q&A] [answered]
Hi everyone,

I'm new to CKAN. My instance in running from the Docker Compose install. 

I want to try some ckan commands, but I don't know where it is located. An example from [documentation](https://docs.ckan.org/en/2.9/maintaining/cli.html) tells:

`/usr/lib/ckan/default/bin/ckan -c /etc/ckan/default/ckan.ini user list`

These file paths don't exist in the CKAN Docker container. Can anyone give me some guidance on how to execute a ckan command in the ckan docker container? I can't find any information about it.

I access the Docker container shell with this command:

`docker exec -it ckan-docker-ckan1 sh`

I use an out-of-the-box installation without any customization.

Hope someone can help me out, 
Frank

**Comment by fpichardom** [comment, marked as answer]
Hi Frank,

The structure in the docker containers is indeed different from what is stated in the documentation for the other installation options. In the containers, the ckan.ini file is located at `/srv/app/ckan.ini.` The containers also come with an environment variable for easy access as `$CKAN_INI.` You can check an example of how it is used to modify the config file at `ckan-docker/ckan/dockerentrypoint.d/01_setup_datapusher.sh` as
 `ckan config-tool $CKAN_INI "ckan.datapusher.api_token=$(ckan -c $CKAN_INI user token add ckan_admin datapusher | tail -n 1 | tr -d '\t')"`

**Reply by frankyhollywood** [reply]
yes that's what I needed to know, thx a lot!
